greedy_algo.py

```diff
@@ -15,54 +15,58 @@ class GreedyReplication:
   def __init__(self):
     self.aggregator = Aggregator() # to retrive server logs
     self.client_set = set([]) # [client_ip, ]
-    self.server_set = set(self.aggregator.retrieve_server_set()) # [server_ip, ]
+    self.server_set = set(util.retrieve_server_list()) # [server_ip, ]
     self.content_set = set([]) # [uuid, ]
     self.access_map = {} # {uuid: {client_ip: num_request}}
     self.replica_map = {} # {uuid: {server_ip: num_replica}}
     self.last_timestamp = 0 # the timestamp of last update
-    self.sample_interval = 1000 # the time interval between two rounds in second
+    self.requests_per_replica = 500
+    self.uuid_to_server = None
+    # self.sample_interval = 1000 # the time interval between two rounds in second
 
   # update client_set, server_set, content_set, access_info
   # and replication status
   # call this function before running greedy algorithm
   def update(self):
+    # clear data
+    content_set = set([])
+    access_map = {}
+    replica_map = {}
+    # update content_set, replica_map
+    for server in self.server_list:
+      file_list = get_file_list_on_server(server)
+      for file_uuid in file_list:
+        content_set.add(file_uuid)
+        if file_uuid not in replica_map:
+          replica_map[file_uuid] = {}
+        if server not in replica_map[file_uuid]:
+          replica_map[file_uuid][server] = 0
+        replica_map[file_uuid][server] += 1
+
     current_timestamp = int(time.time())
-    logs = self.aggregator.get_log_entries(last_timestamp, current_timestamp)
-    # used recently generated logs to update inner data structure
+    logs = self.aggregator.get_log_entries(self.last_timestamp, current_timestamp)
+    # used recently generated logs to update access_map
     for log in logs:
       timestamp, uuid, source, source_uuid, dest, req_type, status, response_size = log.split()
```

Comment:
wait.. doesn't the aggregator already return it in a tuple format instead of a text line?